connect/runtime/src/test/java/org/apache/kafka/connect/integration/TaskHandle.java

```diff
@@ -37,25 +34,20 @@
     private final ConnectorHandle connectorHandle;
     private final AtomicInteger partitionsAssigned = new AtomicInteger(0);
     private final StartAndStopCounter startAndStopCounter = new StartAndStopCounter();
-
-    private CountDownLatch recordsRemainingLatch;
-    private CountDownLatch recordsToCommitLatch;
-    private int expectedRecords = -1;
-    private int expectedCommits = -1;
+    private final RecordLatches recordLatches;
```

Comment:
Step 4: Replaced the duplicate (or very similar) code in `TaskHandle` and `ConnectorHandle` for record writes and commits. Moved all of that logic into the new `RecordLatches` that is used in both `TaskHandle` and `ConnectorHandle`.